src/Controls/src/SourceGen/ProjectItem.cs

```diff
@@ -1,22 +1,101 @@
+using System;
+
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.Maui.Controls.Xaml;
 
 namespace Microsoft.Maui.Controls.SourceGen;
 
-record ProjectItem
+record ProjectItem(AdditionalText AdditionalText, AnalyzerConfigOptions Options)
 {
-	public required AdditionalText AdditionalText { get; init; }
-	private string? _targetPath;
-	public string? TargetPath { get => _targetPath ?? AdditionalText.Path; init => _targetPath = value; }
-	public string? RelativePath { get; init; }
-	public string? ManifestResourceName { get; init; }
-	public required string Kind { get; init; }
-	public string? TargetFramework { get; init; }
-	public required XamlInflator Inflator { get; init; } 
-
-	//bypass attribute check. used for testing
-	public string Configuration { get; internal set; } = "Debug";
-	public bool EnableLineInfo { get; internal set; } = true;
-	public bool EnableDiagnostics { get; internal set; } = false;
-	public string NoWarn { get; internal set; } = "";
+	public string Configuration
+		=> Options.TryGetValue("build_metadata.additionalfiles.Configuration", out var configuration)
+				? configuration
+				: "Debug";
+
+	public bool EnableLineInfo
+	{
+		get
+		{
+			var enableLineInfo = true;
+			if (Options.TryGetValue("build_property.MauiXamlLineInfo", out var lineInfo) && string.Compare(lineInfo, "disable", StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
I don't want to delay work on this with nitpicks, so feel free to ignore this comment. Could this read just `Options.IsDisabled("build_property.MauiXamlLineInfo")` or `!Options.IsEnabled("build_property.MauiXamlLineInfo")`?